File: ButtonTemplateVB/Converters/ResourceToMediaImageConverter.vb
Imports System.Globalization
Imports System.Reflection
Imports System.Resources
Imports System.Windows.Interop
Imports System.Windows.Data
Imports System.Windows.Media.Imaging
Imports System
Imports System.Windows

Namespace Converters

    Public Class ResourceToMediaImageConverter
        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
            If value Is Nothing Then
                Return Nothing
            End If
            Dim Image As System.Drawing.Bitmap
            Image = GetBitmap(value.ToString())
            If Image Is Nothing Then
                Return Nothing
            End If

            Dim bitmap = New System.Drawing.Bitmap(Image)
            Dim BitmapSource = Imaging.CreateBitmapSourceFromHBitmap(bitmap.GetHbitmap(), IntPtr.Zero, Int32Rect.Empty, BitmapSizeOptions.FromEmptyOptions())

            bitmap.Dispose()

            Return BitmapSource
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
        Private Function GetBitmap(ResourceName As String) As System.Drawing.Bitmap
            Dim rm As New ResourceManager("$safeprojectname$.Properties.Resources",
                Assembly.GetExecutingAssembly())
            Try
                Return rm.GetObject(ResourceName)
            Catch ex As ArgumentNullException
                Throw ex
            End Try
        End Function

    End Class
End Namespace

File: ButtonTemplateCS2022/Converters/ResourceToMediaImageConverter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Reflection;
using System.Resources;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;
using System.Windows.Interop;
using System.Windows.Media.Imaging;

namespace $safeprojectname$.Converters
{
    public class ResourceToMediaImageConverter : IValueConverter
    {

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value == null)
                return null;
            var image = GetBitmap(value.ToString());
            if (image == null)
                return null;
            var bitmap = new System.Drawing.Bitmap(image);
            var bitmapSource = Imaging.CreateBitmapSourceFromHBitmap(bitmap.GetHbitmap(),
                                                                                  IntPtr.Zero,
                                                                                  Int32Rect.Empty,
                                                                                  BitmapSizeOptions.FromEmptyOptions()
                  );

            bitmap.Dispose();

            return bitmapSource;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
        private System.Drawing.Bitmap GetBitmap(string ResourceName)
        {
            ResourceManager rm = new ResourceManager("$safeprojectname$.Properties.Resources",
            Assembly.GetExecutingAssembly());
            try
            {
                return (System.Drawing.Bitmap)rm.GetObject(ResourceName);
            }
            catch (ArgumentNullException ex)

            { throw ex; }

        }
    }
}
